### PR TITLE
Adding Linting and checks

### PR DESCRIPTION
**Pull request by lawrencegripper**
#38 - Done
#39 - Done

### CODE REVIEW

**Comment by jjcollinge** [comment]
Just curious; is `make checks` different to `make lint` - and how does `goclangci` handle a `.toml` and `.yml`?

**Comment by lawrencegripper** [comment]
Oh, I'm an idiot didn't realise we had a `lint` already. Yeah they're doing the same things just that the `toml` file enables a few extra behaviours from the default. (Yeah `golangci` is happy with toml or yaml)

**Comment by jjcollinge** [comment]
Ok cool, should we remove the existing `golangci.yml` in favour of this `golangci.toml` once complete?

**Comment by lawrencegripper** [comment]
@jjcollinge fixed up my silliness, fixed up go linting errors and added mdspell - should be good to go if build passes.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 DEV_CONTAINER_TAG:=devcontainer
 
-build: 
+build: checks
```
Comment (jjcollinge) [inline Makefile:3]: `checks` doesn't exist anymore

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Currently this looks to test whether the required basic behaviour is possible. I
 
 ## Aims
 
-Users should be able to deploy the bridge into their cluster with a provider selected like "AzureRMv1.2". It should then:
+Users should be able to deploy the bridge into their cluster with a provider selected like `AzureRMv1.2`. It should then:
 
 1. Generation of mappings TF->CRD should take place at startup/in cluster for ease of use. (Build time step considered but discarded due to added complexity on the user)
```
Comment (jjcollinge) [inline README.md:24]: Change to improve flow:
Generation -> Generate

File: Makefile (hunk 1 above)
Comment (lawrencegripper) [reply Makefile]: Oh man - good spot - trying to do too much and not focusing.

File: README.md (hunk 2 above)
Comment (lawrencegripper) [reply README.md]: Agree, done.

**Comment by lawrencegripper** [comment]
Right should be good to go @jjcollinge (build failed as rebasing on main brought in some failures which are now fixed)

**Pull request merged**